src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py

```diff
@@ -949,7 +949,8 @@ def load_arguments(self, _):
     with self.argument_context('network vnet peering') as c:
         c.argument('virtual_network_name', virtual_network_name_type)
         c.argument('virtual_network_peering_name', options_list=('--name', '-n'), help='The name of the VNet peering.', id_part='child_name_1')
-        c.argument('remote_virtual_network', options_list=('--remote-vnet-id',), help='ID of the remote VNet.')
+        c.argument('remote_virtual_network', options_list=['--remote-vnet', c.deprecate(target='--remote-vnet-id', redirect='--remote-vnet', expiration='2.0.52')], help='Resource ID or name of the remote VNet.')
```

Comment:
`help='Name or ID of the remote VNet.'`

Also, expiration should be a breaking release, so `2.1.0`.  Additionally, the deprecated option should be hidden so it is not discoverable.